### PR TITLE
Service loopback

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Review by aanm — COMMENTED** [review]
>Note: IPv4 only implementation

Does this happens with IPv6?

> FIXME: Needs a dedicated IPv4 address to ensure packets from the local host can never cause clashing tuples.

Where is this IP needed? Can you provide more info regarding this?

**Comment by tgraf** [comment]
IPv6 is also affected. Changes are WIP. We can merge it separately.

Whenever a service is looped back into the same pod, the IP will be used as source. Because we are not rewriting source ports to be unique, the source IP may not be used by anyone else in the context of that host.

**Comment by tgraf** [comment]
@aanm Adapted to reserve an IPv4 address out of the alloc range for service loopback use. This addresses the clash concern mentioned in the previous commit message.

I think we can merge the IPv4 part first and see if this is sufficient and performant and then add the IPv6 part as we have some experience with it.

File: daemon/daemon/daemon.go [hunk 1]
```diff
@@ -422,6 +424,13 @@ func NewDaemon(c *Config) (*Daemon, error) {
 		return nil, err
 	}
 
+	// Allocate IPv4 service loopback IP
```
Comment (aanm) [inline daemon/daemon/daemon.go]: Put this condition `if d.conf.IPv4Enabled`

**Review by aanm — CHANGES_REQUESTED** [review]

**Pull request merged**